src/lib/components/TocScrollSpy/index.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the Licenspe.
+ */
+
+import {ScrollSpy} from '../ScrollSpy';
+
+/**
+ * Element that adds active classes to its children when the user scrolls to
+ * a heading this child points to.
+ * @extends {ScrollSpy}
+ * @final
+ */
+class TocScrollSpy extends ScrollSpy {
+  constructor() {
+    super();
+    this.selector = this.selector = 'h2[id], h3[id]';
+    this.tocActiveClass = 'w-scroll-spy__active';
+    this.tocVisibleClass = 'w-scroll-spy__visible';
+  }
+
+  scrollSpy(headings) {
+    const links = new Map(
+      [...this.querySelectorAll('a')].map((l) => [l.getAttribute('href'), l]),
+    );
+    for (const heading of headings) {
+      const href = `#${heading.target.getAttribute('id')}`;
+      const link = links.get(href);
+
+      if (link) {
+        if (heading.intersectionRatio > 0) {
+          link.classList.add(this.tocVisibleClass);
+          this.previouslyActiveHeading = heading.target.getAttribute('id');
```

Comment:
I'm having a hard time figuring out whether this value is being stored as an instance variable (`this.previouslyActiveHeading`) because it needs to persist across invocations of the `scrollSpy()` callback.

If the value does need to persist across calls to `scrollSpy()`, could you please add a comment explaining how it's being used and what the expectations are?

If the value doesn't need to persist, then it seems like this would make more sense if it were declared as a local variable scoped to the function invocation, instead of an instance variable on `this`.